wcfsetup/install/files/lib/system/user/authentication/password/algorithm/Pbkdf2.class.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace wcf\system\user\authentication\password\algorithm;
+
+use ParagonIE\ConstantTime\Hex;
+use wcf\system\user\authentication\password\IPasswordAlgorithm;
+
+/**
+ * Implementation of the pbkdf2 password algorithm.
+ *
+ * @author      Marcel Werk
+ * @copyright   2001-2023 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.0
+ */
+final class Pbkdf2 implements IPasswordAlgorithm
+{
+    /**
+     * @inheritDoc
+     */
+    public function verify(
+        #[\SensitiveParameter]
+        string $password,
+        string $hash
+    ): bool {
+        $parts = \explode(':', $hash, 5);
```

Comment:
```suggestion
        $parts = \explode(':', $hash, 5);
        if (\count($parts) !== 5) {
            return false;
        }
```
I thought about an `\assert()` but I’m not sure if that makes sense at all.